nucypher/blockchain/eth/utils.py

```diff
@@ -87,45 +91,59 @@ def rpc_endpoint_health_check(endpoint: str, max_drift_seconds: int = 60) -> boo
             timeout=5
         )
     except requests.exceptions.RequestException:
+        LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: network error")
         return False
 
     if response.status_code != 200:
+        LOGGER.debug(
+            f"RPC endpoint {endpoint} is unhealthy: {response.status_code} | {response.text}"
+        )
         return False
 
     try:
         data = response.json()
         if "result" not in data:
+            LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: no response data")
```

Comment:
I think this should be a higher level warning, possibly `warn`?

Reply:
Yeah, that's fair. Maybe I can make this a bit more concise.

Reply:
We may not want to do `warn` on an individual basis - but perhaps rather based on a proportion of overall endpoints that failed, for example if > 50%  of the 10 total available endpoints are unhealthy, then print a warn message, as opposed to printing a warn message for each endpoint every time it is checked and is unhealthy.

Reply:
✔️